rest-server/src/controllers/job.js

```diff
@@ -26,77 +27,42 @@ const load = (req, res, next, jobName) => {
   new Job(jobName, (job, error) => {
     if (error) {
       if (error.message === 'JobNotFound') {
-        if (req.method !== 'PUT') {
-          logger.warn('load job %s error, could not find job', jobName);
-          return res.status(404).json({
-            error: 'JobNotFound',
-            message: `could not find job ${jobName}`,
-          });
+        if (req.method !== 'PUT' && req.method !== 'POST') {
+          error.status = httpStatus.NOT_FOUND;
+          error.message = `could not find job ${jobName}`;
+          next(error);
         }
       } else {
-        logger.warn('internal server error');
-        return res.status(500).json({
-          error: 'InternalServerError',
-          message: 'internal server error',
-        });
+        error.status = httpStatus.INTERNAL_SERVER_ERROR;
+        next(error);
       }
     } else {
-      if (job.jobStatus.state !== 'JOB_NOT_FOUND' && req.method === 'PUT' && req.path === `/${jobName}`) {
-        logger.warn('duplicate job %s', jobName);
-        return res.status(400).json({
-          error: 'DuplicateJobSubmission',
-          message: `job already exists: '${jobName}'`,
-        });
+      if (req.method === 'PUT' && req.path === `/${jobName}` || req.method === 'POST' && req.path === '/') {
```

Comment:
>(req.method [](start = 9, length = 11)

please add a comment: in the future, will remove this check and let Launcher handles this.

Reply:
ok